src/System.Linq/src/System/Linq/Contains.cs

```diff
@@ -26,20 +26,9 @@ public static bool Contains<TSource>(this IEnumerable<TSource> source, TSource v
                 comparer = EqualityComparer<TSource>.Default;
             }
 
-            if (source == null)
-            {
-                throw Error.ArgumentNull(nameof(source));
-            }
-
-            foreach (TSource element in source)
-            {
-                if (comparer.Equals(element, value))
-                {
-                    return true;
-                }
-            }
+            EnumerableHelpers.ThrowIfNull(source, nameof(source));
 
-            return false;
+            return source.Any(element => comparer.Equals(element, value));
```

Comment:
This doesn't look too good... it's adding an additional closure where there wasn't one before.